src/consensus/aft/raft.h

```diff
@@ -2531,5 +2532,14 @@ namespace aft
         }
       }
     }
+
+  public:
+    void update_parameters(kv::ConsensusParameters& params)
+    {
+      CCF_ASSERT_FMT(
+        params.reconfiguration_type != TWO_TRANSACTION || node_client,
+        "Bug; all enclaves that support 2tx reconfig must have node_clients");
+      reconfiguration_type = params.reconfiguration_type;
```

Comment:
Since the consensus logic can be executed by multiple threads (at least when [`worker_threads` is > 0](https://microsoft.github.io/CCF/main/operations/configuration.html#worker-threads)), this feels like it needs to acquired the `state->lock` here?

Reply:
Good point, I agree.

Reply:
Actually, it deadlocks when I do that. The lock is apparently already help during a global hook, although I can't easily find where that's done.

Reply:
Good point. On the primary, it seems that lock is taken in `recv_append_entries_response()` which eventually calls `commit()` via `update_commit()`. We usually write a comment at the top of the function body when it's the case (see e.g. `get_term_history()`)

Reply:
Great, I'll add a similar comment then!